ttnn/cpp/ttnn/operations/moreh/moreh_sgd/moreh_sgd.cpp

```diff
@@ -63,9 +54,9 @@ std::vector<bool> MorehSgd::create_async_return_flag(
     const std::optional<MemoryConfig>& param_out_memory_config,
     const std::optional<MemoryConfig>& momentum_buffer_out_memory_config,
     const std::optional<DeviceComputeKernelConfig>& compute_kernel_config) {
-    if (momentum != 0.0f)
-        return {true, true};
-    return {true, false};
+    return {
+        std::optional<Tensor>(operation::get_workers_for_op_output({param_in, grad}, {momentum_buffer_in})),
```

Comment:
No need to create std::optional. It will be created implicitly. Just create a Tensor or nullopt

Reply:
```C++
 error: incompatible operand types ('Tensor' (aka 'tt::tt_metal::Tensor') and 'const nullopt_t')

        (momentum != 0.0f) ? Tensor(operation::get_workers_for_op_output({param_in, grad}, {momentum_buffer_in})) : std::nullopt};
```
I tried removing `optional`, but it seems that implicit creation is not happening.

Reply:
I see. It would work if that was a single return. But given two things are in the same statement compiler can’t figure out the common denominator. 

Noted. Thank you